### PR TITLE
ARROW-9280: [Rust] [Parquet] Calculate page and column statistics

### PR DESCRIPTION
**Pull request by zeevm**
1. Calculate page and column statistics
2. Use pre-calculated statistics when available to speed-up when writing data from other formats like ORC.

### CODE REVIEW

**Comment by nevi-me** [comment]
Hi @zeevm, may you please kindly rebase (to fix the Rust failures) and open a JIRA for this PR

**Comment by zeevm** [comment]
Opened a Jira.

Rebased but the build still fails

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by nevi-me** [comment]
Thanks @zeevm, the remaining failure can be fixed by running `cargo +stable fmt` from within the rust folder. We use the `rustfmt` from the stable toolchain, to avoid more frequent changes from a nightly one.

File: rust/parquet/src/column/writer.rs
Comment (sunchao) [inline rust/parquet/src/column/writer.rs:284]: Seems this is not used at all?

File: rust/parquet/src/column/writer.rs
Comment (sunchao) [inline rust/parquet/src/column/writer.rs:543]: maybe we should extract this into a util function `update_page_min_max` and similarly for column min/max value.

**Review by sunchao — COMMENTED** [review]
Thanks @zeevm - left some comments.

File: rust/parquet/src/column/writer.rs
Comment (zeevm) [reply rust/parquet/src/column/writer.rs:388]: IIUC the format specifies that the various stats are optional so it seems reasonable to allow the caller to specify only some of the values isn't it?

File: rust/parquet/src/column/writer.rs
Comment (zeevm) [reply rust/parquet/src/column/writer.rs:259]: These are used to track page level stats and write those stats when writing a page and to update the column level stats when writing the page.

Comment (sunchao) [reply rust/parquet/src/column/writer.rs:259]: Yes. But it is not updated nor used at all. Can you double check?

Comment (zeevm) [reply rust/parquet/src/column/writer.rs:259]: Are you specifically referring to 'page_distinct_count', or all 4 page level vars?

Comment (zeevm) [reply rust/parquet/src/column/writer.rs:259]: All of them are used, page_distinct_count isn't being calculated in this PR though, probably a following PR

Comment (zeevm) [reply rust/parquet/src/column/writer.rs:259]: They're used here:

flush_data_pages()
make_typed_statistics()
update_page_min_max()

Comment (sunchao) [reply rust/parquet/src/column/writer.rs:259]: Yes I meant `page_distinct_count`. It is fine to do this in a follow upo.

**Review by sunchao — APPROVED** [review]
LGTM

File: rust/parquet/src/column/writer.rs
Comment (zeevm) [reply rust/parquet/src/column/writer.rs:259]: How do I merge? Do I need to ask for merge/write permissions to complete the PR?

**Comment by zeevm** [comment]
How do I complete/merge the PR, do I need to request write access or will it be completed by a maintainer?

**Comment by sunchao** [comment]
@zeevm once approved, a committer will help merge this. Seems the PR now is a little messed up, can you clean it up so I can merge it?

**Pull request closed without merging**
